frontend/src/metabase/search/containers/SearchApp.jsx

```diff
@@ -25,172 +30,90 @@ import {
   SearchRoot,
 } from "./SearchApp.styled";
 
-const PAGE_SIZE = 50;
-
-const SEARCH_FILTERS = [
-  {
-    name: t`Apps`,
-    filter: "app",
-    icon: "star",
-  },
-  {
-    name: t`Dashboards`,
-    filter: "dashboard",
-    icon: "dashboard",
-  },
-  {
-    name: t`Collections`,
-    filter: "collection",
-    icon: "folder",
-  },
-  {
-    name: t`Databases`,
-    filter: "database",
-    icon: "database",
-  },
-  {
-    name: t`Models`,
-    filter: "dataset",
-    icon: "model",
-  },
-  {
-    name: t`Raw Tables`,
-    filter: "table",
-    icon: "table",
-  },
-  {
-    name: t`Questions`,
-    filter: "card",
-    icon: "bar",
-  },
-  {
-    name: t`Pulses`,
-    filter: "pulse",
-    icon: "pulse",
-  },
-  {
-    name: t`Metrics`,
-    filter: "metric",
-    icon: "sum",
-  },
-  {
-    name: t`Segments`,
-    filter: "segment",
-    icon: "segment",
-  },
-];
-
 export default function SearchApp({ location }) {
   const { handleNextPage, handlePreviousPage, setPage, page } = usePagination();
-  const [filter, setFilter] = useState(location.query.type);
 
-  const handleFilterChange = filterItem => {
-    setFilter(filterItem && filterItem.filter);
-    setPage(0);
-  };
+  const searchText = useMemo(
+    () => getSearchTextFromLocation(location),
+    [location],
+  );
 
-  const query = {
-    q: location.query.q,
-    limit: PAGE_SIZE,
-    offset: PAGE_SIZE * page,
-  };
+  const searchFilters = useMemo(() => {
+    return getFiltersFromLocation(location);
+  }, [location]);
+
+  const [selectedSidebarType, setSelectedSidebarType] = useState(null);
+
+  useEffect(() => {
+    setSelectedSidebarType(null);
+  }, [searchText, searchFilters]);
+
+  const query = useMemo(
```

Comment:
I don't think you need to memoize `query`. From my brief investigation, `query` should be memoized under the hood with entity loader. This is probably for convenience so that people using entity loading don't have to memoize every query pass to it.

https://github.com/metabase/metabase/blob/2181d118b74b00e26aa88bac51b7ea1def5bb96d/frontend/src/metabase/entities/containers/EntityListLoader.jsx#L74-L80

Reply:
Ah nice catch!